features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/ImageButtonMapper.kt

```diff
@@ -6,97 +6,74 @@
-                view,
-                id,
-                bounds,
-                shapeStyle,
-                border,
-                drawable
-            )
-            wireframes.add(imageWireframe)
+        if (backgroundDrawable != null && shapeStyle == null && border == null) {
+            // try to resolve background as a wireframe
+            @Suppress("ThreadSafety") // TODO REPLAY-1861 caller thread of .map is unknown?
+            imageWireframeHelper.createImageWireframe(
+                view = view,
+                0,
+                x = bounds.x,
+                y = bounds.y,
+                width = view.width.densityNormalized(density).toLong(),
+                height = view.height.densityNormalized(density).toLong(),
+                drawable = backgroundDrawable.constantState?.newDrawable(resources),
+                shapeStyle = null,
+                border = null
+            )?.let {
+                wireframes.add(it)
+            }
         }
 
-        return wireframes
-    }
-
-    // region internal
+        val (scaledDrawableWidth, scaledDrawableHeight) =
+            base64Serializer.getDrawableScaledDimensions(view, foregroundDrawable, density)
 
-    private fun resolveImageWireframe(
-        view: ImageButton,
-        id: Long,
-        bounds: GlobalBounds,
-        shapeStyle: MobileSegment.ShapeStyle?,
-        border: MobileSegment.ShapeBorder?,
-        drawable: Drawable
-    ): MobileSegment.Wireframe.ImageWireframe {
-        val mimeType = getWebPMimeType()
-        val displayMetrics = view.resources.displayMetrics
-        val applicationContext = view.context.applicationContext
-
-        val imageWireframe =
-            MobileSegment.Wireframe.ImageWireframe(
-                id = id,
-                x = bounds.x,
-                y = bounds.y,
-                width = bounds.width,
-                height = bounds.height,
-                shapeStyle = shapeStyle,
-                border = border,
-                base64 = "",
-                mimeType = mimeType,
-                isEmpty = true
-            )
+        val centerX = (bounds.x + view.width.densityNormalized(density) / 2) - (scaledDrawableWidth / 2)
+        val centerY = (bounds.y + view.height.densityNormalized(density) / 2) - (scaledDrawableHeight / 2)
 
+        // resolve foreground
         @Suppress("ThreadSafety") // TODO REPLAY-1861 caller thread of .map is unknown?
-        handleBitmap(
-            applicationContext = applicationContext,
-            displayMetrics = displayMetrics,
-            drawable = drawable,
-            imageWireframe = imageWireframe
-        )
+        imageWireframeHelper.createImageWireframe(
+            view = view,
+            index = wireframes.size,
+            x = centerX,
+            y = centerY,
+            width = scaledDrawableWidth,
+            height = scaledDrawableHeight,
+            drawable = foregroundDrawable.constantState?.newDrawable(resources),
```

Comment:
I have a question here? If this is an `ImageButton` why do you resolve it as `backround` + `foreground` wireframe and not just take one `screenshot` of the whole element ?

Reply:
It's calling the drawable constructor per drawable type so I think not necessarily so heavy. Regarding taking a screenshot, we could go that route but it would mean having two different strategies on the mappers instead of having the same logic for all going forward. Also potentially it would be more resource intensive to take screenshots because you're implicitly creating a new bitmap, while you could potentially have a bitmapDrawable on the imagebutton and get the bitmap for free in the wireframe.